server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/OffHeapStarTreeBuilder.java

```diff
@@ -242,6 +260,11 @@ public StarTreeDocument getStarTreeDocument(int docId) throws IOException {
         return starTreeDocumentFileManager.readStarTreeDocument(docId, true);
     }
 
+    @Override
+    public StarTreeDocument getStarTreeDocumentForCreatingDocValues(int docId) throws IOException {
```

Comment:
Why do we need this ? we can directly use `getStarTreeDocument(docId)` ?

Reply:
I think it was introduced in the previous PRs, let me get back if this required. looks okay to swap for now.